### PR TITLE
Update configuration for dev/ci containers to work with nvidia-docker 2.0

### PR DESCRIPTION
**Pull request by dalg24**

### CODE REVIEW

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-version: '2'
+version: '2.3'
```
[the remark this answers is not in the document]
Comment (dalg24) [reply docker/docker-compose.yml]: I get an error when I run with `version: '3'` and nvidia-docker says to use 2.3

File: docker/nvidia/setup_nvidia_docker_compose.py [hunk 2]
```diff
@@ -111,18 +30,19 @@ def get(self, endpoint):
         # in that case we would have to append install of CUDA toolchain to the Dockerfile specified in the build section
         assert 'build' not in base_config['services'][service]
 
+# see https://github.com/NVIDIA/nvidia-docker/wiki/Frequently-Asked-Questions#do-you-support-docker-compose
+# NOTE consider setting nvidia runtime as the default in which case I don't think the version of docker-compose will matter any more
+assert compose_file_version == '2.3'
```
Comment (dalg24) [inline docker/nvidia/setup_nvidia_docker_compose.py]: @Rombur See that assertion

File: docker/docker-compose.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Rombur) [reply docker/docker-compose.yml]: OK

**Comment by dalg24** [comment]
Test are passing for me locally.. Closing and reopening :/

**Pull request merged**